staging/src/k8s.io/client-go/dynamic/simple.go

```diff
@@ -225,6 +229,7 @@ func (c *dynamicResourceClient) DeleteCollection(ctx context.Context, opts metav
 	result := c.client.client.
 		Delete().
 		AbsPath(c.makeURLSegments("")...).
+		SetHeader("Content-Type", runtime.ContentTypeJSON).
```

Comment:
Good question. We want the dynamic client to behave the same as a typed client. In typed client case, the parameter passed to Body() is binary [data](https://github.com/kubernetes/kubernetes/blob/2584a7c0bb27a5029c3ac24b496a20da58edca35/staging/src/k8s.io/client-go/kubernetes/typed/core/v1/pod.go#L198). Due to this [code](https://github.com/kubernetes/kubernetes/blob/2584a7c0bb27a5029c3ac24b496a20da58edca35/staging/src/k8s.io/client-go/rest/request.go#L457),  the typed client won't set the 'content-type' header, so I think it's ok to not set the header in dynamic client.

We perhaps should set the 'content-type' header to `json` for the patch methods for both typed and dynamic clients, but that would be another PR, and will need more careful review because it's potentially a breaking change.

Reply:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/dynamic/simple.go#L284-L294